### PR TITLE
BaSP-TG-105: FE - Employee Profile Bugs

### PR DESCRIPTION
**Pull request by Johsua30**
**Changes made:**
- Added the .allow('') property to the fields not present at creation(address, picture, dni, dateBirth) and to the password.
- The password field will by default be empty and allow an empty string as a valid input. If the field does not contain an empty string when submit is pressed it is takes as if the field was edited.
- Added a default value of '' to address, picture, dni, dateBirth in the creation thunk to save an empty string instead of undefined or null.

This PR is using changes from https://github.com/BaSP-m2022/alexis-trackgenix-server/pull/52 in the backend.

**Intended result:**
On creation the fields not present in the sign up form will contain an empty string and will not be required upon editing the employee.
The password field in the edit profile form will contain an empty string and any different input will be taken as an attempt to edit the password. If all checks pass the inserted password is saved.

### CODE REVIEW

**Review by agusting19 — APPROVED** [review]
Good job, everything looks nice!

File: src/Components/Employee/Profile/Edit/index.jsx [hunk 1]
```diff
@@ -87,17 +88,30 @@ const EmployeeFormEdit = ({ employeeEdit, closeModalForm }) => {
       setResponse({ message: "There haven't been any changes", data: {}, error: true });
```
Comment (iaracriscenti) [inline src/Components/Employee/Profile/Edit/index.jsx:88]: Line 86 could be changed to 
` (data.dateBirth?.toString() == new Date(employeeEdit.dateBirth) || data.dateBirth === null)` due dateBirth can have a null value now and It'll break if the ? is not added after .toString()

File: src/Components/Employee/Profile/Edit/index.jsx [hunk 2]
```diff
@@ -116,7 +130,7 @@ const EmployeeFormEdit = ({ employeeEdit, closeModalForm }) => {
       lastName: employeeEdit.lastName,
       phone: employeeEdit.phone,
       email: employeeEdit.email,
-      password: employeeEdit.password,
+      password: '',
       address: employeeEdit.address,
       picture: employeeEdit.picture,
       dni: employeeEdit.dni,
```
Comment (iaracriscenti) [inline src/Components/Employee/Profile/Edit/index.jsx:136]: Line 137 could be change to `dateBirth: employeeEdit.dateBirth ? employeeEdit.dateBirth.slice(0, 10) : null` for the same reason, if it has a null value, keep it that way, if not, apply the slice method

**Review by iaracriscenti — APPROVED** [review]
Bugs fixed, great job!

**Review by tinchorova — APPROVED** [review]

**Pull request merged**